### DISCUSSION TITLE
Monitoring for successful backup independently

### DISCUSSION THREAD

**Discussion by Alestrix** [borgbackup/borg#7729] [Q&A] [answered]
Hi,

I monitor many of the services I run in the house (influxdb, Grafana, various GUIs, access points, smart plugs, solar inverter, ...) using uptime Kuma. So there's no fancy checkmk or zabbix or the like available, but Kuma (running redundantly on a VM and a Pi, respectively) notifies me via instant messenger when something's wrong.

I also have a daily cron job backing up various paths from my NAS to a remote location which supports borg, but no arbitrary scripts (Hetzner storage VM).

Has anyone of you implemented a simple monitoring solution to check that the latest backup is not older than 24 hours which does not depend on the system doing the backup job? Maybe even using something in combination with Kuma? If so I'd love to hear about it!

Cheers
Alex

**Comment by garret** [comment, marked as answer]
Isn't what [Healthchecks.io](https://healthchecks.io/) is doing?

**Reply by Alestrix** [reply]
Thanks, that's not exactly the way I thought I'd solve it, but it's a solution to my task at hand! If my backup job runs into an error I won't send a "health-ping" (or if my backup VM is toast :smile:) and I'll get a notification. Maybe I'll check the status through Kuma (for the one-pane-of-glass feeling) or I'll just have healthchecks.io send the notification directly to my messenger service.

**Comment by ThomasWaldmann** [comment]
It's not easy to do that server side (just looking at the repo), because borg's threat model is that the server is untrusted. Repos usually are encrypted, so you are not able to see when the last successful backup was.

You can see that last "activity" though, if you look at the time stamp of the files, e.g. the repo index.

**Reply by Alestrix** [reply]
I cannot do it server side anyway (see original post). It's a pure storage VM that supports rsync, sftp, borg and a few other protocols, but I do not have shell access to it.

**Reply by RonnyPfannschmidt** [reply]
Just let the backup script ping the Monitoring System after success?

**Reply by Alestrix** [reply]
Uptime Kuma doesn't support this. And as mentioned I don't run a full fledged monitoring solution.
